packages/rpc-transport/src/json-rpc.ts

```diff
@@ -23,7 +23,9 @@ function createPendingRpcRequest<TRpcMethods, TResponse>(
             if ('error' in response) {
                 throw new SolanaJsonRpcError(response.error);
             } else {
-                return (responseTransformer ? responseTransformer(response.result) : response.result) as TResponse;
+                return (
+                    responseTransformer ? responseTransformer(response.result, methodName) : response.result
```

Comment:
- Why wouldn't we provide the entire response to the transformer?
- Would it be possible to wrap the method name, the payload and the URL in a request object so that the entire request can also be provided to the transformer?

Reply:
> * Why wouldn't we provide the entire response to the transformer?

We can, but I think we're only shedding `jsonrpc` and `id`.
Check out a [response payload](https://docs.solana.com/api/http#getaccountinfo) in the docs.

> * Would it be possible to wrap the method name, the payload and the URL in a request object so that the entire request can also be provided to the transformer?

I believe so, yes!

Reply:
The reason I'm saying all this is I think this should be comparable to [Axios interceptors](https://axios-http.com/docs/interceptors) (or equivalent) where we have no opinion over what could be transformed and we're just like "here's a request I'm about to send, do whatever you want with it" and same with the response. 🙂

Reply:
So, in my opinion I think `createJsonRpcApi(..)` should be designed to create APIs that adhere to the [JSON RPC spec](https://www.jsonrpc.org/specification), without deviation. So, to me, it makes sense to leave part of the request and response unchanged, since modifying `request.parameters` or `response.result` still adheres to the JSON RPC interface.

It seems like you probably have the Helius example in mind from #1740 but here, I think this function would _not_ be used to implement the custom transport you referenced there. What do you think?

Reply:
Yeah that's a fair point actually. So you're allowing the library to transform anything that can be transformed without affecting the JSON RPC specs. I like it!

Am I right to say that, in order to support a `urlTransformer` on the API side, we would need something like a `requestTransformer` on the Transport side? Similarly to the `responseTransformer`, the `requestTransformer` would only allow you to edit the components of the requests that don't affect the JSON RPC specs. Namely, the parameters and the URL.

Reply:
> Am I right to say that, in order to support a `urlTransformer` on the API side, we would need something like a `requestTransformer` on the Transport side? Similarly to the `responseTransformer`, the `requestTransformer` would only allow you to edit the components of the requests that don't affect the JSON RPC specs. Namely, the parameters and the URL.

We _could_ do this in theory, but I can't decide if I think it's something we _should_ support.

Are you thinking you'd be able to plug in a `requestTransformer` that, for example, takes all of the parameters and the method name from the POST body and maps them to URL parameters?

As an aside, I'm also not sure if the Helius endpoints for example are GETs instead of POSTs.

Reply:
Yeah that's what I'm thinking. I guess it all depends what is technically valid in the JSON RPC specs. Is it okay to use dynamic URLs? Is it okay to use different HTTP methods? If not, then maybe we shouldn't mess with it. If it is valid according to the specs though, then I strongly believe we should offer that level of flexibility.

Reply:
I can't see anything in the spec that says definitively one way or the other. But it does _seem_ like the intention is a request body.